### ISSUE TITLE
Missing max_cells param for scanpy.api.pp.filter_genes()

### ISSUE THREAD

**Issue by jorvis** [scverse/scanpy#77]
The [filter_cells()](http://scanpy.readthedocs.io/en/latest/api/scanpy.api.pp.filter_cells.html#scanpy.api.pp.filter_cells) provides parameters min_genes and max_genes, but the corresponding [filter_genes()](http://scanpy.readthedocs.io/en/latest/api/scanpy.api.pp.filter_genes.html#scanpy.api.pp.filter_genes) has only min_cells.  Any objection to adding max_cells for completeness?  I have a user who has requested this already, which is why I ran into it.

**Comment by falexwolf** [comment]
Of course, let's add this. Do you want to do it or shall I?

**Comment by jorvis** [comment]
Happy to try

**Comment by jorvis** [comment]
There's my initial attempt, but it fails with:

```pytb
File "/usr/local/lib/python3.6/dist-packages/scanpy/preprocessing/simple.py", line 169, in filter_genes
    adata.var['n_cells'] = number
  File "/usr/local/lib/python3.6/dist-packages/pandas/core/frame.py", line 2519, in __setitem__
    self._set_item(key, value)
  File "/usr/local/lib/python3.6/dist-packages/pandas/core/frame.py", line 2585, in _set_item
    value = self._sanitize_column(key, value)
  File "/usr/local/lib/python3.6/dist-packages/pandas/core/frame.py", line 2760, in _sanitize_column
    value = _sanitize_index(value, self.index, copy=False)
  File "/usr/local/lib/python3.6/dist-packages/pandas/core/series.py", line 3121, in _sanitize_index
    raise ValueError('Length of values does not match length of ' 'index')
ValueError: Length of values does not match length of index
```

If it isn't obvious to you what's wrong I'll return to it after my conference this weekend.  Every hour is critical right now.  :)

**Comment by adelq** [comment]
Is this issue complete? It seems like the bug has been fixed after 96890730972162aa531c3289b38ad728a7585c85

**Comment by jorvis** [comment]
@falexwolf reported it as a fix, yes.  I will close.
